Regex/FoundationUtils.swift

```diff
@@ -15,21 +15,20 @@
 //===----------------------------------------------------------------------===//
 
 #if !os(Linux)
-    //here we use NSRegularExpression
-    import Foundation
+import Foundation
+#endif
     
-    enum InvalidRangeError : ErrorType {
```

Comment:
Moved extension from NSRange to type alias GroupRange. And make it allowed for all platforms